pyHS100/pyHS100.py

```diff
@@ -333,7 +341,18 @@ def mac(self, mac):
         :param str mac: mac in hexadecimal with colons, e.g. 01:23:45:67:89:ab
         :raises SmartPlugException: on error
         """
-        self._query_helper("system", "set_mac_addr", {"mac": mac})
+        info = self.sys_info
+
+        if 'mac' in info:
+            self._query_helper("system", "set_mac_addr", {"mac": mac})
+        elif 'mic_mac' in info:
+            self._query_helper(
+                "system",
+                "set_mac_addr",
+                {"mic_mac": mac.replace(':', '')}
```

Comment:
well i noticed that the mic_mac does not have :'s in the mac address in the first PR i did i posted the actual data that came from a user that has an LB100 and in there the mic_mac does not have :'s

and it can be set up for a single call but this is an if, elif. so in the event there is a 3rd kind wandering around it will not try to set it incorrectly. it would be more code to check if there is payload data before sending the query. however you want to set it up is up to you